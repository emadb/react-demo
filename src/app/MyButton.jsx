export default function MyButton({text, onClick}) {
  return <button className="btn btn-default" onClick={onClick} >{text}</button>
}