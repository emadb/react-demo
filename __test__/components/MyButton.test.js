import React from 'react'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import MyButton from '../../src/app/MyButton'

describe('<MyButton />', () => {
  
  test('just render with default props', () => {
    const component = renderer.create(<MyButton />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('click should call fn', done => {
    const wrapper = shallow(<MyButton text="Click me" onClick={() => done()} />)
    wrapper.find('button').simulate('click')
  })

})