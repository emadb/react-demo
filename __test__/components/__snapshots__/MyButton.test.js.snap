// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyButton /> just render with default props 1`] = `
<button
  className="btn btn-default"
  onClick={undefined}
/>
`;
